frontend/server/handlers/artifacts.ts

```diff
@@ -239,6 +254,111 @@ function getGCSArtifactHandler(options: { key: string; bucket: string }, peek: n
   };
 }
 
+function getVolumeArtifactsHandler(options: { bucket: string; key: string }, peek: number = 0) {
+  const { key, bucket } = options;
+  return async (req: Request, res: Response) => {
+    let filePath;
+    let volumeMount;
+    try {
+      // get ml-pipeline-ui pod namespace
+      const SERVER_NAMESPACE = serverInfo.getServerNamespace();
+      // get ml-pipeline-ui pod name
+      const POD_NAME = serverInfo.getServerPodName();
+
+      // only check when POD_NAME and serverNamespace can be obtained
+      if (POD_NAME && SERVER_NAMESPACE) {
+        const [pod, err] = await k8sHelper.getPod(POD_NAME, SERVER_NAMESPACE);
```

Comment:
nit: Can we cache the result in some module? Because pod should be immutable, we don't need to query k8s api server each time.
e.g. we can add a method `k8sHelper.getHostPod()`, intention is clear, we can add caching and it's easy to mock in tests.

Reply:
Great suggestion!

Reply:
I add getHostPod into `helpers/server-info.ts`. if add it into `k8sHelper `, I found it's difficult to mock `k8sHelper ` function.